tests/test_parts_manager.py

```diff
@@ -529,12 +463,12 @@ def test_handle_thinking_delta_no_vendor_id_with_existing_thinking_part():
     manager = ModelResponsePartsManager()
 
     # Add a thinking part first
-    event = manager.handle_thinking_delta(vendor_part_id='first', content='initial thought', signature=None)
+    event = next(manager.handle_thinking_delta(vendor_part_id='first', content='initial thought', signature=None))
     assert isinstance(event, PartStartEvent)
     assert event.index == 0
 
     # Now add another thinking delta with no vendor_part_id - should update the latest thinking part
-    event = manager.handle_thinking_delta(vendor_part_id=None, content=' more', signature=None)
+    event = next(manager.handle_thinking_delta(vendor_part_id=None, content=' more', signature=None))
```

Comment:
this test tests a `ThinkingPart` that gets created with a valid `vendor_part_id` and is later updated with a `vendor_part_id=None`